test/e2e/framework/framework.go

```diff
@@ -132,6 +131,7 @@ func (f *Framework) SetupNatsOperator() error {
 				},
 			},
 			RestartPolicy: v1.RestartPolicyNever,
```

Comment:
For different PR but thinking might be better to change to `RestartPolicyAlways` in case they exit 1 for some reason, right now the operator does replace the pods only when they are missing.